### ISSUE TITLE
Implementation details of CLIP guidance

### ISSUE THREAD

**Issue by ug-kim** [ashawkey/stable-dreamfusion#257]
I have some questions about CLIP guidance option.

If I put both image and text prompts, CLIP guidance is applied.

1. At make-it-3d, CLIP loss is measured between the denoised output of novel view rendering and reference image.
However, in your code you measured between the original novel view rendering and reference image.
Is it for the convenience of implementation?

2. At make-it-3d, CLIP loss is measured only between images. Is this an empirical choice?

```python
  def train_step(self, clip_z, pred_rgb, grad_scale=10, **kwargs):

      image_z = self.clip_model.encode_image(self.aug(pred_rgb))
      image_z = image_z / image_z.norm(dim=-1, keepdim=True) # normalize features

      loss = 0
      if 'image' in clip_z:
          loss = loss - (image_z * clip_z['image']).sum(-1).mean()
      
      if 'text' in clip_z:
          loss = loss - (image_z * clip_z['text']).sum(-1).mean()

      loss = loss * grad_scale

      return loss
```

**Comment by ashawkey** [comment]
@ug-kim Hi,
1. As the denoising step is `no_grad`, the gradient propagated from the denoised images should be the same as the original image (only the scale is different), so I just use the original images to save computation.
2. In their released code they do clip loss with both images and text, so I follow their implementation.

That being said, The results of text + image input still need some tuning, so these choices may be changed later.
